fml/trace_event.h

```diff
@@ -33,19 +33,25 @@
 
 #ifndef TRACE_EVENT_HIDE_MACROS
 
+#define __FML__TOKEN_CAT__(x, y) x##y
+#define __FML__TOKEN_CAT__2(x, y) __FML__TOKEN_CAT__(x, y)
+#define __FML__AUTO_TRACE_END(name)                                  \
+  ::fml::tracing::ScopedInstantEnd __FML__TOKEN_CAT__2(__trace_end_, \
+                                                       __LINE__)(name);
```

Comment:
Consider adding a comment explaining the need for concatenation voodoo to deal with the macro expansion for `__LINE__`.

Reply:
The old implementation just didn't work. I actually don't know what to say other than this is how to do concatenation of line numbers. Happy to add more documentation in a subsequent patch.